homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,255 @@
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup platform."""
+    station_id = config.get(CONF_STATION_ID)
+    name = config.get(CONF_NAME)
+
+    probe = ZamgData(station_id=station_id, logger=_LOGGER)
+
+    sensors = [ZAMGWeather(probe, variable, name)
+               for variable in config[CONF_MONITORED_CONDITIONS]]
+
+    add_devices(sensors, True)
+
+
+# ------------------------------------------------------------------------
+class ZAMGWeather(Entity):
+    """
+    I am a weather wrapper for a specific station and a specific attribute.
+
+    Multiple instances (one for each condition) will refer to the same
+    probe, so things will only get fetched once.
+    """
+
+    attribution = ATTRIBUTION
+    temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, probe, variable, name):
+        """Init condition sensor."""
+        self.probe = probe
+        self.client_name = name
+        self.variable = variable
+
+    def update(self):
+        """Delegate update to probe."""
+        self.probe.update()
+
+    @property
+    def name(self):
+        """Build name of sensor."""
+        return "%s_%s" % (self.client_name, self.variable)
+
+    @property
+    def state(self):
+        """Return state."""
+        return self.probe.get_data(self.variable)
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_WEATHER_ATTRIBUTION: self.attribution,
+            "friendly_name": SENSOR_TYPES[self.variable][0],
+            "station": self.probe.get_data('station_name'),
+            "updated": "%s %s" % (self.probe.get_data('update_date'),
+                                  self.probe.get_data('update_time'))
+
+        }
+
+
+# ------------------------------------------------------------------------
+class ZamgData(object):
+    """
+    I represent weather data for a specific site.
+
+    From the web site:
+
+    Sie beinhalten neben Stationsnummer, Stationsname, Seehöhe der Station,
+    Messdatum und Messzeit (Lokalzeit) die meteorologischen Messwerte von
+    Temperatur, Taupunkt, relative Luftfeuchtigkeit, Richtung und
+    Geschwindigkeit des Windmittels und der Windspitze, Niederschlagssumme
+    der letzten Stunde, Luftdruck reduziert auf Meeresniveau und Luftdruck
+    auf Stationsniveau sowie die Sonnenscheindauer der letzten Stunde (in
+    Prozent). Die Messstationen, die diese Daten liefern, sind über das
+    Bundesgebiet verteilt und beinhalten alle Landeshauptstädte sowie
+    die wichtigsten Bergstationen.
+    """
+
+    API_URL = "http://www.zamg.ac.at/ogd/"
+
+    API_FIELDS = {
+        v[2]: (k, v[3])
+        for k, v in SENSOR_TYPES.items()
+    }
+
+    API_HEADERS = {
+        'User-Agent': 'home-assistant.zamg/' + __version__,
+    }
+
+    def __init__(self, logger, station_id):
+        """Initialize the probe."""
+        self._logger = logger
```

Comment:
No, that's intentional, so one could test that sensor class by injecting a mock logger. I don't like to touch module or global scoped vars in multiple places if it can be avoided.